File: src/ServerGuard.Agent/Config/AgentConfig.cs
using System.Text.Json.Serialization;

namespace ServerGuard.Agent.Config;
public class AgentConfig
{
    public Guid AgentId { get; private set; }
    public bool IsCpuEnabled { get; private set; }
    public bool IsGpuEnabled { get; private set; }
    public bool IsMemoryEnabled { get; private set; }
    public bool IsMotherboardEnabled { get; private set; }
    public bool IsControllerEnabled { get; private set; }
    public bool IsNetworkEnabled { get; private set; }
    public bool IsStorageEnabled { get; private set; }
    public int CollectEverySeconds { get; private set; }
    public string RabbitUsername { get; private set; } = default!;
    public string RabbitPassword { get; private set; } = default!;
    public string RabbitUrl { get; private set; } = default!;
    public string RabbitExchange { get; private set; } = default!;
    public string ApiKey { get; private set; } = default!;

    public AgentConfig()
    {
    }

    [JsonConstructor]
    public AgentConfig(
        Guid agentId,
        bool isCpuEnabled,
        bool isGpuEnabled,
        bool isMemoryEnabled,
        bool isMotherboardEnabled,
        bool isControllerEnabled,
        bool isNetworkEnabled,
        bool isStorageEnabled,
        int collectEverySeconds,
        string rabbitUsername,
        string rabbitPassword,
        string rabbitUrl,
        string rabbitExchange,
        string apiKey)
    {
        AgentId = agentId;
        IsCpuEnabled = isCpuEnabled;
        IsGpuEnabled = isGpuEnabled;
        IsMemoryEnabled = isMemoryEnabled;
        IsMotherboardEnabled = isMotherboardEnabled;
        IsControllerEnabled = isControllerEnabled;
        IsNetworkEnabled = isNetworkEnabled;
        IsStorageEnabled = isStorageEnabled;
        CollectEverySeconds = collectEverySeconds;
        RabbitUsername = rabbitUsername;
        RabbitPassword = rabbitPassword;
        RabbitUrl = rabbitUrl;
        RabbitExchange = rabbitExchange;
        ApiKey = apiKey;
    }

    public override bool Equals(object? obj)
    {
        if (obj is AgentConfig options)
        {
            return AgentId == options.AgentId
                && IsCpuEnabled == options.IsCpuEnabled
                && IsGpuEnabled == options.IsGpuEnabled
                && IsMemoryEnabled == options.IsMemoryEnabled
                && IsMotherboardEnabled == options.IsMotherboardEnabled
                && IsControllerEnabled == options.IsControllerEnabled
                && IsNetworkEnabled == options.IsNetworkEnabled
                && IsStorageEnabled == options.IsStorageEnabled
                && CollectEverySeconds == options.CollectEverySeconds
                && RabbitUsername == options.RabbitUsername
                && RabbitPassword == options.RabbitPassword
                && RabbitUrl == options.RabbitUrl
                && RabbitExchange == options.RabbitExchange
                && ApiKey == options.ApiKey;
        }
        return base.Equals(obj);
    }

    public override int GetHashCode()
    {
        return base.GetHashCode();
    }
}

File: ServerGuard.Api/src/main/java/com/mmieczkowski/serverguard/agent/model/AgentConfig.java
package com.mmieczkowski.serverguard.agent.model;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;

import java.io.Serializable;

@Embeddable
public class AgentConfig implements Serializable {

    public AgentConfig(boolean isCpuEnabled,
                       boolean isGpuEnabled,
                       boolean isMemoryEnabled,
                       boolean isMotherboardEnabled,
                       boolean isControllerEnabled,
                       boolean isNetworkEnabled,
                       boolean isStorageEnabled,
                       int collectEverySeconds) {
        this.isCpuEnabled = isCpuEnabled;
        this.isGpuEnabled = isGpuEnabled;
        this.isMemoryEnabled = isMemoryEnabled;
        this.isMotherboardEnabled = isMotherboardEnabled;
        this.isControllerEnabled = isControllerEnabled;
        this.isNetworkEnabled = isNetworkEnabled;
        this.isStorageEnabled = isStorageEnabled;
        this.collectEverySeconds = collectEverySeconds;
    }

    public AgentConfig() {
    }

    @Column(nullable = false)
    private boolean isCpuEnabled;

    @Column(nullable = false)
    private boolean isGpuEnabled;

    @Column(nullable = false)
    private boolean isMemoryEnabled;

    @Column(nullable = false)
    private boolean isMotherboardEnabled;

    @Column(nullable = false)
    private boolean isControllerEnabled;

    @Column(nullable = false)
    private boolean isNetworkEnabled;

    @Column(nullable = false)
    private boolean isStorageEnabled;

    @Column(nullable = false)
    private int collectEverySeconds;

    @Column
    private String apiKey;

    public boolean isCpuEnabled() {
        return this.isCpuEnabled;
    }

    public boolean isGpuEnabled() {
        return this.isGpuEnabled;
    }

    public boolean isMemoryEnabled() {
        return this.isMemoryEnabled;
    }

    public boolean isMotherboardEnabled() {
        return this.isMotherboardEnabled;
    }

    public boolean isControllerEnabled() {
        return this.isControllerEnabled;
    }

    public boolean isNetworkEnabled() {
        return this.isNetworkEnabled;
    }

    public boolean isStorageEnabled() {
        return this.isStorageEnabled;
    }

    public int getCollectEverySeconds() {
        return this.collectEverySeconds;
    }

    public String getApiKey() {
        return this.apiKey;
    }

    public void setApiKey(String apiKey) {
        this.apiKey = apiKey;
    }

    public String toString() {
        return "AgentConfig(isCpuEnabled=" + this.isCpuEnabled() + ", isGpuEnabled=" + this.isGpuEnabled() + ", isMemoryEnabled=" + this.isMemoryEnabled() + ", isMotherboardEnabled=" + this.isMotherboardEnabled() + ", isControllerEnabled=" + this.isControllerEnabled() + ", isNetworkEnabled=" + this.isNetworkEnabled() + ", isStorageEnabled=" + this.isStorageEnabled() + ", collectEverySeconds=" + this.getCollectEverySeconds() + ", apiKey=" + this.getApiKey() + ")";
    }

    public boolean equals(final Object o) {
        if (o == this) return true;
        if (!(o instanceof AgentConfig)) return false;
        final AgentConfig other = (AgentConfig) o;
        if (!other.canEqual((Object) this)) return false;
        if (this.isCpuEnabled() != other.isCpuEnabled()) return false;
        if (this.isGpuEnabled() != other.isGpuEnabled()) return false;
        if (this.isMemoryEnabled() != other.isMemoryEnabled()) return false;
        if (this.isMotherboardEnabled() != other.isMotherboardEnabled()) return false;
        if (this.isControllerEnabled() != other.isControllerEnabled()) return false;
        if (this.isNetworkEnabled() != other.isNetworkEnabled()) return false;
        if (this.isStorageEnabled() != other.isStorageEnabled()) return false;
        if (this.getCollectEverySeconds() != other.getCollectEverySeconds()) return false;
        final Object this$apiKey = this.getApiKey();
        final Object other$apiKey = other.getApiKey();
        if (this$apiKey == null ? other$apiKey != null : !this$apiKey.equals(other$apiKey)) return false;
        return true;
    }

    protected boolean canEqual(final Object other) {
        return other instanceof AgentConfig;
    }

    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = result * PRIME + (this.isCpuEnabled() ? 79 : 97);
        result = result * PRIME + (this.isGpuEnabled() ? 79 : 97);
        result = result * PRIME + (this.isMemoryEnabled() ? 79 : 97);
        result = result * PRIME + (this.isMotherboardEnabled() ? 79 : 97);
        result = result * PRIME + (this.isControllerEnabled() ? 79 : 97);
        result = result * PRIME + (this.isNetworkEnabled() ? 79 : 97);
        result = result * PRIME + (this.isStorageEnabled() ? 79 : 97);
        result = result * PRIME + this.getCollectEverySeconds();
        final Object $apiKey = this.getApiKey();
        result = result * PRIME + ($apiKey == null ? 43 : $apiKey.hashCode());
        return result;
    }
}
